tests/playwright/src/specs/kube-contexts.spec.ts

```diff
@@ -0,0 +1,130 @@
+/**********************************************************************
+ * Copyright (C) 2024 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ***********************************************************************/
+
+import fs from 'node:fs';
+import path from 'node:path';
+import { fileURLToPath } from 'node:url';
+
+import { KubeContextPage } from '../model/pages/kube-context-page';
+import { PreferencesPage } from '../model/pages/preferences-page';
+import { expect as playExpect, test } from '../utility/fixtures';
+
+const testContexts = ['context-1', 'context-2', 'context-3'];
+
+const __filename = fileURLToPath(import.meta.url);
+const __dirname = path.dirname(__filename);
+
+test.beforeAll(async ({ runner, welcomePage }) => {
+  runner.setVideoAndTraceName('kube-context-e2e');
+
+  // copy testing kubeconfig file to the expected location
+  const kubeConfigPathSrc = path.resolve(__dirname, '..', '..', 'resources', 'test-kube-config');
+  const kubeConfigPathDst = path.resolve(__dirname, '..', '..', 'resources', 'kube-config');
+  fs.copyFileSync(kubeConfigPathSrc, kubeConfigPathDst);
+
+  await welcomePage.handleWelcomePage(true);
+});
+
+test.afterAll(async ({ runner }) => {
+  test.setTimeout(120000);
+  await runner.close();
+});
+
+test.describe.serial('Verification of kube context management', () => {
+  test('Load custom kubeconfig in Preferences', async ({ navigationBar }) => {
+    test.setTimeout(12000);
```

Comment:
Timeout is in ms, so this timeout is 12 sec. Default is 60 sec, you sure you want to make it shorter?